src/mscorlib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -185,6 +190,15 @@ public Assembly LoadFromAssemblyName(AssemblyName assemblyName)
             Assembly assembly = Load(assemblyName);
             if (assembly == null)
             {
+                // Invoke the AssemblyResolve event callbacks if wired up
+                if (AssemblyResolve != null)
+                {
+                    assembly = AssemblyResolve(this, assemblyName);
```

Comment:
This does not work for multiple subscribes - this call will call all subscribers, and return whatever the last one returned.

Instead, you need to iterate over the subscribers manually and stop once one returns non-null value.